src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -488,6 +600,12 @@ public function addAllowedValues($option, $allowedValues)
             throw new AccessException('Allowed values cannot be added from a lazy option or normalizer.');
         }
 
+        if ($this->isNested($option)) {
+            @trigger_error(sprintf('The "%s" method should not be used with nested options. A failed attempt occurred with the option "%" for values %s', __METHOD__, $option, $this->formatValues($allowedValues)), E_USER_WARNING);
```

Comment:
this doesn't seem to be a deprecation notice, so exceptions should be used, right?